internal/common/logging/stacktrace.go

```diff
@@ -5,18 +5,34 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+const Stacktrace = "Stacktrace"
+
 // Unexported but considered part of the stable interface of pkg/errors.
 type stackTracer interface {
 	StackTrace() errors.StackTrace
 }
 
+// Unexported but considered part of the stable interface of pkg/errors.
+type causer interface {
+	Cause() error
+}
+
 // WithStacktrace returns a new logrus.Entry obtained by adding error information and, if available, a stack trace
 // as fields to the provided logrus.Entry.
 func WithStacktrace(logger *logrus.Entry, err error) *logrus.Entry {
 	logger = logger.WithError(err)
+	stack := ExtractStack(err)
+	if stack != nil {
+		logger = logger.WithField(Stacktrace, stack)
+	}
+	return logger
+}
+
+func ExtractStack(err error) errors.StackTrace {
```

Comment:
Are you sure this is correct?
https://stackoverflow.com/a/54387284

Reply:
Yes- the issue is that the solution you linked to in stackoverflow loses the stack if the error has been wrapped.  As an example:
```
func TestLogStack(t *testing.T){
	err := errors.New("some error")
	wrappedErr := errors.WithMessage(err, "some message")

	if st, ok := wrappedErr.(stackTracer); ok {
		logrus.WithField("stacktrace", st.StackTrace()).Infof("stackoverflow solution- error with stack")
	} else{
		logrus.Infof("stackoverflow solution- error without stack")
	}

	logrus.WithField(logging.Stacktrace, logging.ExtractStack(wrappedErr)).Infof("my solution with stack")
}
```
...
```
time="2023-03-07T11:37:48Z" level=info msg="stackoverflow solution- error without stack"
time="2023-03-07T11:37:48Z" level=info msg="my solution with stack" stacktrace="[queue_test.go:373 testing.go:1439 asm_amd64.s:1571]"
```